src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -999,13 +999,11 @@ private static unsafe BlittableJsonReaderObject ReadInternal<T>(TransactionOpera
 
             var info = await ReplicationUtils.GetTcpInfoAsync(url, null, "Cluster", certificate);
 
-            var tcpInfo = new Uri(info.Url);
-            var tcpClient = new TcpClient();
+            var tcpClient = TcpUtils.NewTcpClient(_parent.TcpConnectionTimeout);
             Stream stream = null;
             try
             {
-                TcpUtils.SetTimeouts(tcpClient, _parent.TcpConnectionTimeout);
```

Comment:
Note that now we call SetTimeouts out of the try..catch, and it might be an issue for the RC release if it will throw NRE.